.github/workflows/transports-ci.yml

```
@@ -0,0 +1,194 @@
+      # Checkout with full history and tags for version management
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: main
+          fetch-depth: 0 # Full history needed for git tag operations
+          fetch-tags: true # Ensure all tags are available
+          token: ${{ secrets.GH_TOKEN }} # Personal token for pushing tags
+
+      # Set up Go environment for building binaries and managing dependencies
+      - name: Set up Go
+        uses: actions/setup-go@v4
```

actionlint (1.7.7)

the runner of "actions/setup-go@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)